kythe/cxx/indexer/cxx/GraphObserver.h

```diff
@@ -970,6 +970,13 @@ class GraphObserver {
   virtual void recordDeprecated(const NodeId& NodeId,
                                 const llvm::StringRef& Advice) {}
 
+  /// \brief Records a diagnostic at the given source range
+  /// \param Range The source range
+  /// \param Signature The signature to use for the diagnostic VName
+  /// \param Message The diagnostic message
+  virtual void recordDiagnostic(const Range& Range, std::string_view Signature,
+                                std::string_view Message) {}
```

Comment:
minor: This should probably use `llvm::StringRef` (or we should switch every method to use `std::string_view`)